content/rust/concepts/data-types/data-types.md

```diff
@@ -0,0 +1,81 @@
+---
+Title: 'Data-Types'
+Description: 'A brief description.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Conditionals'
+  - 'Control Flow'
+CatalogContent:
+  - 'rust-for-programmers'
+  - 'paths/computer-science'
+---
+
+**Data types** serve to define the nature of data a variable can store. Rust employs static typing, ensuring that variable types are known at compile time, aiding in the detection of potential errors before runtime. These data types are broadly categorized into two groups: primitive types and compound types.
+
+## Primitive types || scala types
+
+Scala types represent a single value.
+
+- Integer
+- Floating-Point
+- Boolean
+- Character
+
+## Compound Types
+
+compund types represent multiple values and can group them together.
+
+- tuples
+- arrays
```

Comment:
do same for these too smal desc will be good. Also check the rust entries see if there are tuples and arrays entry link them here.